prism/src/main/java/org/apache/falcon/resource/extensions/ExtensionManager.java

```diff
@@ -0,0 +1,432 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.falcon.resource.extensions;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.POST;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.falcon.FalconException;
+import org.apache.falcon.FalconWebException;
+import org.apache.falcon.entity.v0.Entity;
+import org.apache.falcon.entity.v0.process.Process;
+import org.apache.falcon.entity.v0.feed.Feed;
+import org.apache.falcon.entity.v0.cluster.Cluster;
+import org.apache.falcon.extensions.Extension;
+import org.apache.falcon.extensions.ExtensionProperties;
+import org.apache.falcon.extensions.store.ExtensionStore;
+import org.apache.falcon.resource.APIResult;
+import org.apache.falcon.resource.AbstractSchedulableEntityManager;
+import org.apache.falcon.resource.EntityList;
+import org.apache.falcon.resource.ExtensionJobList;
+import org.apache.falcon.resource.ExtensionInstanceList;
+import org.apache.falcon.resource.InstancesResult;
+import org.apache.falcon.util.DeploymentUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Jersey Resource for extension job operations.
+ */
+@Path("extension")
+public class ExtensionManager extends AbstractSchedulableEntityManager {
+    public static final Logger LOG = LoggerFactory.getLogger(ExtensionManager.class);
+
+    private static final ExtensionStore STORE = ExtensionStore.get();
+    private static final String RESULTS = "extensions";
+    private static final String TOTAL_RESULTS = "totalResults";
+    private static final String README = "README";
+    private static final String EXTENSION_PROPERTY_JSON_SUFFIX = "-properties.json";
+    private static final String SHORT_DESCRIPTION = "shortDescription";
+
+    private static final String TRUSTED_EXTENSION = "Trusted extension";
+    private static final String CUSTOM_EXTENSION = "Custom extension";
```

Comment:
I think these are used by extension repository Rest API implementation. Can we remove these so that this Jira has code only related to job management for anyone reviewing or viewing the changes later.